scripts/install/auto-install.sh

```diff
@@ -0,0 +1,245 @@
+#!/bin/bash
+# auto-install.sh installs git if it isn't installed, then clones POCS and
+# installs its dependencies.
+#
+# To fetch this script from github and execute it immediately,
+# run these commands:
+#
+# export POCS_GITHUB_USER="panoptes"
+# export POCS_BRANCH="develop"
+# BASE_RAW_POCS_URL="https://raw.githubusercontent.com/${POCS_GITHUB_USER}/POCS/${POCS_BRANCH}"
+# export AUTO_INSTALL_RAW_URL="${BASE_RAW_POCS_URL}/scripts/install/auto-install.sh"
+# wget -q -O - "${AUTO_INSTALL_RAW_URL}" | bash
+
+################################################################################
+# Functions COPIED from install-helper-functions.sh
+# Print the disk location of the first arg.
+function safe_which() {
+  type -p "${1}" || /bin/true
+}
+
+# Print a separator bar of # characters.
+function echo_bar() {
+  local terminal_width="${COLUMNS}"
+  if [ -z "${terminal_width}" ] && [ -n "${TERM}" ]
+  then
+    if [[ -n "$(safe_which tput)" ]]
+    then
+      terminal_width="$(tput cols)"
+    elif [[ -n "$(safe_which resize)" ]]
+    then
+      terminal_width="$(resize 2>/dev/null | grep COLUMNS= | cut -d= -f2)"
+    elif [[ -n "$(safe_which stty)" ]]
+    then
+      terminal_width="$(stty size 2>/dev/null | cut '-d ' -f2)"
+    fi
+  fi
+  printf "%${terminal_width:-80}s\n" | tr ' ' '#'
+}
+
+################################################################################
+
+function do_sudo() {
+  if [ "$(id -u -n)" == "root" ] ; then
+    echo "Running ${*}"
+    "$@"
+  else
+    echo <<ENDOFMESSAGE
+Running:
+    sudo ${*}
+You may be prompted for your password.
+ENDOFMESSAGE
+    (set -x ; sudo "$@")
+  fi
+}
+
+function clone_or_update() {
+  local -r REPO_DIR="${1}"
+  local -r REPO_URL="${2}"
+  local -r ORIGIN_NAME="${3}"
+  local -r BRANCH="${4}"
+
+  echo_bar
+
+  if [ ! -d "${REPO_DIR}/.git" ]
+  then
+    if [ -d "${REPO_DIR}" ]
+    then
+      echo 2> "
+The directory (${REPO_DIR}) already exists, but doesn't appear
+to be a valid git repository. Please remove it (or move it out
+of the way) and re-run this script.
+  "
+      exit 1
+    fi
+    echo "
+Cloning ${REPO_URL} into ${REPO_DIR}
+"
+    (set -x ; git clone --origin "${ORIGIN_NAME}" "${REPO_URL}" "${REPO_DIR}")
+    cd "${REPO_DIR}"
+    git checkout "${BRANCH}"
+  else
+    echo "
+Pulling the latest software into the worktree at ${REPO_DIR}.
+"
+    cd "${REPO_DIR}"
+    git fetch --all
+    git checkout "${BRANCH}"
+    git pull
```

Comment:
If they have already set up the repo they might have `origin` set to their repo and this pull might not necessarily be doing what we want. Or they might have files in staging, in which case this will fail.

I think if the dir is there and a repo we just do nothing other than maybe print a message reminding them to update. Thoughts?

Reply:
Yeah, clearly if there are already files staged then we need to avoid messing with the repo.
Of course, if they have a repo, they don't need this auto-install script, and can run install-dependencies.sh themselves.

I'll adjust the script to stop if there is a POCS or PAWS repo.